megatron/core/utils.py

```diff
@@ -0,0 +1,60 @@
+from functools import reduce
```

Comment:
We don't have `megatron/core`, so this code needs added to `megatron/utils.py` along with updated imports